packages/core/src/compiler/Style.ts

```diff
@@ -1219,50 +1239,52 @@ const matchStyApplyToSubApply = (
   varEnv: Env,
   styRel: RelPred<A> | SelExpr<A>,
   subRel: ApplyPredicate<A> | SubExpr<A>
-): Subst | undefined => {
+): Subst[] => {
   // Predicate Applications
   if (styRel.tag === "RelPred" && subRel.tag === "ApplyPredicate") {
     // If names do not match up, this is an invalid matching. No substitution.
     if (subRel.name.value !== styRel.name.value) {
-      return undefined;
+      return [];
     }
-    let rSubst = matchStyArgsToSubArgs(
+
+    // Consider the original version
+    const rSubst1 = matchStyArgsToSubArgs(
       styTypeMap,
       subTypeMap,
       varEnv,
       styRel.args,
       subRel.args
     );
-    if (rSubst === undefined) {
-      // check symmetry
-      const predicateDecl = varEnv.predicates.get(subRel.name.value);
-      if (predicateDecl && predicateDecl.symmetric) {
-        // Flip arguments
-        const flippedStyArgs = [styRel.args[1], styRel.args[0]];
-        rSubst = matchStyArgsToSubArgs(
-          styTypeMap,
-          subTypeMap,
-          varEnv,
-          flippedStyArgs,
-          subRel.args
-        );
-      }
+
+    // Consider the symmetric, flipped-argument version
```

Comment:
The refactoring in this function is not super clear to me. Looks like the original version already considered flipped args for symmetry. What are you changing here at a high level?

Also `rSubst1` and `rSubst2` are not very meaningful names.

Reply:
Yup I can explain that.

The original algorithm is:
```
rSubst1 = get subst
if (rSubst1 is empty) {
  if (isSymmetric) {
    flip arguments
    rSubst2 = get subst
  }
}
```

whereas the new algorithm is
```
rSubst1 = getSubst
if (isSymmetric) {
  flip arguments
  rSubst2 = getSubst
}
```

In particular, the original algorithm considers the symmetric version if and only if
* the predicate is symmetric, and
* the original version of the predicate does not match

On the other hand, the new algorithm considers the symmetric version if the predicate is symmetric, regardless of whether the original version of the predicate matches.

Reply:
Sure I can change the names

Reply:
Suppose a predicate is symmetric. The original algorithm checks for the symmetric version of the predicate IF the original version fails to match. The new algorithm tries the symmetric version regardless.

This is captured by the fact that we removed `if (rSubst === undefined)`

Reply:
Yeah makes a lot of sense. Basically this is the place where you got rid of the original short-circuiting. Thanks!